resources/js/features/root/layout/Navbar.tsx

```diff
@@ -1,73 +1,130 @@
     <aside class={cx(s.sidebar)}>
-      <FullLogo />
-      <nav class="flex-1">
-        <Show when={params.facilityUrl}>
-          <For each={NAVBAR_ITEMS}>
-            {(item) => (
-              <A
-                href={item.hrefFn({location, params})}
-                class="mb-2 p-4 rounded-lg flex flex-row items-center gap-3 hover:bg-white"
-                activeClass="bg-white"
-              >
-                <item.Icon size="25" />
-                <span>{item.title}</span>
-              </A>
-            )}
-          </For>
+      <div class={cx("flex flex-row justify-between py-4 px-8 bg-inherit")}>
+        <img src="/img/memo_logo.svg" class="h-14" />
+        <img src="/img/cpd_children_logo.svg" class="h-12" />
+      </div>
+      <nav class={cx("flex-1 px-8 py-4 overflow-y-auto", s.navScroll)}>
+        <Show when={facilityUrl()}>
+          <NavigationSection
+            facilityUrl={facilityUrl()}
+            roles={["verified"]}
+            items={sectionItems().looseFacilityItems}
+          />
+        </Show>
+        <NavigationSection items={sectionItems().looseItems} />
+        <NavigationSection roles={["globalAdmin"]} items={sectionItems().system} title="System" />
+        <Show when={facilityUrl()}>
+          <NavigationSection
+            facilityUrl={facilityUrl()}
+            roles={["facilityAdmin"]}
+            items={sectionItems().facility}
+            title="Placówka"
+          />
+          <NavigationSection
+            facilityUrl={facilityUrl()}
+            roles={["facilityStaff"]}
+            items={sectionItems().work}
+            title="Moja praca"
+          />
         </Show>
-        <hr class="my-4" />
-        <A
-          href="/admin"
-          class="mb-2 p-4 rounded-lg flex flex-row items-center gap-3 hover:bg-white"
-          activeClass="bg-white"
-        >
-          <HiSolidUserGroup size="25" />
-          <span>Administracja</span>
-        </A>
-        <A
-          href="/help"
-          class="mb-2 p-4 rounded-lg flex flex-row items-center gap-3 hover:bg-white"
-          activeClass="bg-white"
-        >
-          <HiSolidInformationCircle size="25" />
-          <span>Pomoc</span>
-        </A>
       </nav>
     </aside>
   );
 };
 
-export type HrefFnArgs = {
-  params: {facilityUrl?: string};
-  location: Location;
-};
-
-export type NavbarItem = {
-  Icon: IconTypes;
-  hrefFn: (args: HrefFnArgs) => AnchorProps["href"];
-  title: string;
-};
-
-const NAVBAR_ITEMS: NavbarItem[] = [
-  {
-    Icon: HiSolidUserGroup,
-    hrefFn: ({params: {facilityUrl}}) => {
-      if (facilityUrl) return `/${facilityUrl}/admin`;
-      return "/help";
+const getSectionItems = (
+  facilityUrl?: string,
+): {
+  system: NavigationSectionProps["items"];
+  facility: NavigationSectionProps["items"];
+  work: NavigationSectionProps["items"];
+  looseFacilityItems: NavigationSectionProps["items"];
+  looseItems: NavigationSectionProps["items"];
+} => ({
+  system: [
```

Comment:
Ta lista to nawet nie tak do końca się pokrywa z tą w Routerze. Na przykład tam chyba jest facilityId/home czy coś. Trzeba to albo uspójnić, albo nawet ogarnąć coś do robienia tych rzeczy jakby bardziej równolegle. Chociaż pewnie bez sensu, nie będzie aż tyle tych stron. Ale uspójnienie by się przydało, można by mieć jakoś wszystkie URLe gdzieś w jednym miejscu, chociażby jako ileś stałych z poszczególnymi członami. Coś jak:
```ts
{
  admin: {
    _self: "admin",
    facilities: "facilities",
    users: "users",
  },
  facility: {
    _self: "{facilityUrl}",
    ...
  },
}
```
No i potem te w routerze sklejać chamsko ``/${URL.admin._self}/${URL.admin.facilities}``, no a tutaj z jakimś replacem na facilityUrlu. Taki pomysł ale można pewnie lepiej. Ale wtedy przynajmniej byłoby wiadomo gdzie jest autorytatywna lista URLi, bo teraz to nie wiadomo.

Reply:
urle w menu i tak nie mogą się 100% pokrywać z urlami w configu route'ów. W menu obsługujemy tylko konkretne route'y, a konfiguracja będzie o wiele bardziej złożona, im więcej podstron wygenerujemy.

Reply:
No tak, w routerze może być więcej niż w menu, ale wszystko co jest w menu powinno być w routerze. Teraz to nie tylko jest bałagan ale jeszcze nawet trudno jest porównać te dwa miejsca. Moim zdaniem wyciągnięcie tego w jedno miejsce miałoby sens w pewnym momencie, no ale powiedzmy że niekoniecznie teraz.

Czasem mam taki dzień że coś mogę zrobić, o ile to nie jest zbyt ambitne. To brzmi jak zadanie na taki dzień :) No wiedz powiedzmy że może na razie zostać, chociaż miło jakbyś uspójnił te URLe na teraz, żeby nie było śmieci na starcie w tym routerze.

Reply:
zostawię tak jak jest. Z routerem jest taki problem, że narzuca pewną strukturę i BARDZO nie chce mi się tego uspójniać z navbarem :D